flink-table-store-core/src/main/java/org/apache/flink/table/store/file/io/RollingFileWriter.java

```diff
@@ -36,36 +38,53 @@
  */
 public class RollingFileWriter<T, R> implements FileWriter<T, List<R>> {
 
-    private final Supplier<? extends FileWriter<T, R>> writerFactory;
+    private static final Logger LOG = LoggerFactory.getLogger(RollingFileWriter.class);
+
+    private final Supplier<? extends SingleFileWriter<T, R>> writerFactory;
     private final long targetFileSize;
-    private final List<FileWriter<T, R>> openedWriters;
+    private final List<SingleFileWriter<T, R>> openedWriters;
     private final List<R> results;
 
-    private FileWriter<T, R> currentWriter = null;
+    private SingleFileWriter<T, R> currentWriter = null;
     private long lengthOfClosedFiles = 0L;
     private long recordCount = 0;
     private boolean closed = false;
 
     public RollingFileWriter(
-            Supplier<? extends FileWriter<T, R>> writerFactory, long targetFileSize) {
+            Supplier<? extends SingleFileWriter<T, R>> writerFactory, long targetFileSize) {
         this.writerFactory = writerFactory;
         this.targetFileSize = targetFileSize;
         this.openedWriters = new ArrayList<>();
         this.results = new ArrayList<>();
     }
 
+    @VisibleForTesting
+    public long targetFileSize() {
+        return targetFileSize;
+    }
+
     @Override
     public void write(T row) throws IOException {
-        // Open the current writer if write the first record or roll over happen before.
-        if (currentWriter == null) {
-            openCurrentWriter();
-        }
+        try {
+            // Open the current writer if write the first record or roll over happen before.
+            if (currentWriter == null) {
+                openCurrentWriter();
+            }
 
-        currentWriter.write(row);
-        recordCount += 1;
+            currentWriter.write(row);
+            recordCount += 1;
 
-        if (currentWriter.length() >= targetFileSize) {
-            closeCurrentWriter();
+            if (currentWriter.length() >= targetFileSize) {
+                closeCurrentWriter();
+            }
+        } catch (Throwable e) {
+            LOG.warn(
+                    "Exception occurs when writing file "
+                            + (currentWriter == null ? null : currentWriter.path())
+                            + ". Cleaning up.",
+                    e);
+            abort();
```

Comment:
Add document to `FileWriter.write`, clear file by itself now when exception in write.